### PR TITLE
Immunefi 11461

### PR DESCRIPTION
**Pull request by Mouzayan**
Added require statements in `originationController initializeLoan()` function to ensure that the signer and the side in the signature match.
Created test in `originationController.ts` to confirm that the code will revert if there is a mismatch between signer and side.

To run the `originationController.ts` tests:
`npx hardhat test test/OriginationController.ts`

To run the entire test suite use:
`npx hardhat test `

### CODE REVIEW

File: contracts/OriginationController.sol [hunk 1]
```diff
@@ -174,6 +174,12 @@ contract OriginationController is
 
         (bytes32 sighash, address externalSigner) = recoverTokenSignature(loanTerms, sig, nonce, neededSide);
 
+        if (externalSigner == borrower) {
```
Comment (kkennis) [inline contracts/OriginationController.sol:177]: Do we also need this logic in `initializeLoanWithItems`? Seems like it should maybe go inside `_validateCounterparties` to cover the most bases

Comment (kkennis) [reply contracts/OriginationController.sol:177]: Also: revert with a custom error here, as opposed to using `require`? to be consistent with the rest of the file

File: test/Attack.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
+import chai, { expect } from "chai";
```
Comment (kkennis) [inline test/Attack.ts:1]: If we fix the bug, these tests will fail right? Should we repurpose this test, and put it in `OriginationController.ts`, also making sure that we check for failure via a revert as opposed to the test passing?

**Review by kkennis — COMMENTED** [review]
It looks like a lot of the apecoin staking stuff is in this branch, which it doesn't need to be? Seems like maybe the root branch was the wrong one

**Comment by Mouzayan** [comment]
> It looks like a lot of the apecoin staking stuff is in this branch, which it doesn't need to be? Seems like maybe the root branch was the wrong one

Removed.

File: test/Attack.ts (hunk 2 above)
Comment (Mouzayan) [reply test/Attack.ts:1]: This test will not fail after fixing the bug because of its socially engineered flow:

- the signature is [created](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L178) by the lender and specifies the borrower for `Side`
- the vault is initialized by the lender and is [transferred](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L190) to the borrower
- borrower approves bundleId of aforementioned vault to originationController
- borrower gives [approve](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L195) to lender to sign and initialize loans on their behalf -- hence the signature created by the lender, specifying the Borrow as side is OK because of this approve 
-  [initializeLoan](https://github.com/arcadexyz/v2-contracts/blob/980d793682fee9e9a139b54d94b0bca1a789ae35/test/Attack.ts#L205) is run and `_validateCounterparties` is called, where `Side` was set to `B` by the lender but the borrower sanctioned them to do so with the `approve`, so no `revert` occurs.

I think we should remove this test. It is too much of an edge case.
Do you agree?

**Comment by kkennis** [comment]
I think you still need to remove the changes to `IAssetVault.sol` and `AssetVault.sol`, right? I don't see anything in this PR that would touch the vault

**Comment by kkennis** [comment]
Also even though you removed the staking files, the commit structure is still out of whack -  you still have commits from the apecoin branch.

I'm guessing you just removed the files in order to clean them up, but it's really the old commits that need to be cleaned up - this should be a clean branch off main, with no cruft from other branches.

**Comment by Mouzayan** [comment]
PR replaced with branch `fix-immunefi-11461` PR and closed.
